evalml/pipelines/binary_classification_pipeline.py

```diff
@@ -37,10 +42,14 @@ def _predict(self, X, objective=None):
         if self.threshold is None:
             return self._component_graph.predict(X)
         ypred_proba = self.predict_proba(X).to_dataframe()
+        predictions = self._predict_with_objective(X, ypred_proba, objective)
+        return infer_feature_types(predictions)
+
+    def _predict_with_objective(self, X, ypred_proba, objective):
```

Comment:
Helper function that takes in `predict_proba`, so we don't have to recalculate it for each objective.